src/python/pants/core/goals/export.py

```diff
@@ -203,11 +260,77 @@ async def export(
         raise UnrecognizedResolveNamesError(
             unexported_resolves,
             all_valid_resolve_names,
+            all_known_bin_names,
             description_of_origin="the option --export-resolve",
         )
 
     return Export(exit_code=0)
 
 
+async def link_exported_executables(
+    build_root: BuildRoot,
+    environment: EnvironmentVars,
+    output_dir: str,
+    export_results: list[ExportResult],
+):
+    """Link the exported executables to the `bin` dir.
+
+    Multiple resolves might export the same executable. This will export the first one only but
+    track the collision.
+    """
+    bin_dir = Path(build_root.path, output_dir, "bin")
+    safe_mkdir(bin_dir)
+
+    exported_bins_by_exporting_resolve: dict[str, list[str]] = defaultdict(list)
+    link_requests = []
+    for result in export_results:
+        for exported_bin in result.exported_binaries:
+            exported_bins_by_exporting_resolve[exported_bin.name].append(
+                result.resolve or result.description
+            )
+            if len(exported_bins_by_exporting_resolve[exported_bin.name]) > 1:
+                continue
+
+            ln_bin = shutil.which("ln")
+            if not ln_bin:
+                # I think ln_bin missing won't happen, but the error message otherwise is "No such file or directory (os error 2)" which isn't helpful
+                raise RuntimeError(
+                    "Could not locate `ln` bin to link exported binaries to the `bin` dir"
+                )
```

Comment:
This seems fine (i.e. I don't think we need to block merging this, but could instead refactor later), but did you consider the option of creating a digest with symlinks in it and including that in the "main" `workspace.write_digest(...)` call? Rather than shell out to symlinking.

Reply:
yeah good point. I lifted the idea from the python backend's `export` implementation, which uses a PostProcessingCommand to symlink the venv in.
It probably makes more sense to collect all the symlinks and write them out, though. I'll see how hard that is

Reply:
yeah that's a lot simpler, thanks for the suggestion! I've kept it as a separate `write_digest` to keep the locality of the processing, there's a lot going on.

Reply:
also the symlinks are absolute symlinks, relative symlinks complained about the symlink not being in the subtree. Should be fine?